src/tests/build.sh

```diff
@@ -268,156 +70,70 @@ build_Tests()
-        generate_layout
-    fi
-}
-
-build_MSBuild_projects()
-{
-    subDirectoryName="$1"
-    shift
-    projectName="$1"
-    shift
-    stepName="$1"
-    shift
-    extraBuildParameters=("$@")
-
     # Set up directories and file names
-    __BuildLogRootName="$subDirectoryName"
     __BuildLog="$__LogsDir/${__BuildLogRootName}.${__TargetOS}.${__BuildArch}.${__BuildType}.log"
     __BuildWrn="$__LogsDir/${__BuildLogRootName}.${__TargetOS}.${__BuildArch}.${__BuildType}.wrn"
     __BuildErr="$__LogsDir/${__BuildLogRootName}.${__TargetOS}.${__BuildArch}.${__BuildType}.err"
+    __BuildBinLog="$__LogsDir/${__BuildLogRootName}.${__TargetOS}.${__BuildArch}.${__BuildType}.binlog"
+    __msbuildLog="\"/flp:Verbosity=normal;LogFile=${__BuildLog}\""
+    __msbuildWrn="\"/flp1:WarningsOnly;LogFile=${__BuildWrn}\""
+    __msbuildErr="\"/flp2:ErrorsOnly;LogFile=${__BuildErr}\""
+
+    # Uncomment the line below when instrumenting Linux builds to produce binlogs;
+    # they seem to be too lengthy to be enabled by default.
+    __msbuildBinLog="\"/bl:${__BuildBinLog}\""
+
+    # Export properties as environment variables for the MSBuild scripts to use
+    export __TestDir
+    export __TestIntermediatesDir
+    export __NativeTestIntermediatesDir
+    export __BinDir
+    export __TestBinDir
+    export __SkipManaged
+    export __SkipGenerateLayout
+    export __SkipTestWrappers
+    export __BuildTestProject
+    export __BuildTestDir
+    export __BuildTestTree
+    export __RuntimeFlavor
+    export __CopyNativeProjectsAfterCombinedTestBuild
+    export __CopyNativeTestBinaries
+    export __Priority
+    export __DoCrossgen2
+    export __CreatePerfMap
+    export __CompositeBuildMode
+    export __BuildTestWrappersOnly
+    export __GenerateLayoutOnly
+    export __TestBuildMode
+    export __MonoAot
+    export __MonoFullAot
+    export __MonoBinDir
+    export __MsgPrefix
+    export __ErrMsgPrefix
+    export __Exclude
 
-    if [[ "$subDirectoryName" == "Tests_Managed" ]]; then
-        # Execute msbuild managed test build in stages - workaround for excessive data retention in MSBuild ConfigCache
-        # See https://github.com/Microsoft/msbuild/issues/2993
-
-        # __SkipPackageRestore and __SkipTargetingPackBuild used  to control build by tests/src/dirs.proj
-        __SkipPackageRestore=false
-        __SkipTargetingPackBuild=false
-        __NumberOfTestGroups=3
-
-        __AppendToLog=false
-
-        if [[ -n "$__priority1" ]]; then
-            __NumberOfTestGroups=10
-        fi
+    # Generate build command
+    buildArgs=("$__RepoRootDir/src/tests/build.proj")
+    buildArgs+=("/t:TestBuild")
+    buildArgs+=("${__CommonMSBuildArgs}")
+    buildArgs+=("/maxcpucount")
+    buildArgs+=("${__msbuildLog}" "${__msbuildWrn}" "${__msbuildErr}" "${__msbuildBinLog}")
+    buildArgs+=("/p:NUMBER_OF_PROCESSORS=${__NumProc}")
+    buildArgs+=("${__UnprocessedBuildArgs[@]}")
+
+    # Disable warnAsError - coreclr issue 19922
```

Comment:
Can we change this comment to have the url to the correct issue?

Reply:
Thanks for spotting, fixed in updated iteration.